### PR TITLE
Postdeploy relayer

### LINKED ISSUE #272: Add Camelot Relayer to predeployment
Due to Solidity version collisions, the Camelot relayer and factory (and supporting contracts) has be moved to the od-relayer repo compiled at Sol 0.7.6:  https://github.com/open-dollar/od-relayer

Next steps:
- create pre-deployment script for the relayer
- add appropriate contract addresses to the Registry.s.sol to be used in protocol deployment

Security:
- add access control to relayer and factory permissioned to the DAO
- behaviour should reflect access control standard of protocol

### PR DESCRIPTION
**Pull request by daopunk**
closes #272

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -318,23 +263,16 @@ contract DeployAnvil is GoerliParams, Deploy {
     IBaseOracle _rethOracle =
       denominatedOracleFactory.deployDenominatedOracle(_rethETHOracle, _wstethUSDPriceFeed, false);
 
-    // MAGIC
```
Comment (pi0neerpat) [inline script/Deploy.s.sol:321]: Why is MAGIC removed?

File: script/Registry.s.sol [hunk 2]
```diff
@@ -58,49 +51,37 @@ uint256 constant MIN_DELAY = 3 days; // timelock for governor
...
+uint256 constant MAINNET_SALT_PROTOCOLTOKEN = 1;
+uint256 constant MAINNET_SALT_VAULT721 = 1;
 address constant MAINNET_CREATE2_FACTORY = address(0);
 
+// Camelot Relayer (pre-deployed @ sol 0.7.6)
+address constant MAINNET_CAMELOT_RELAYER_FACTORY = address(0);
+address constant MAINNET_CAMELOT_RELAYER = address(0);
+
+// Camelot Protocol
+address constant CAMELOT_AMM_FACTORY = 0x1a3c9B1d2F0529D97f2afC5136Cc23e58f1FD35B;
```
Comment (pi0neerpat) [inline script/Registry.s.sol:64]: Is the naming convention here different from testnet? I don't see this one above

File: script/create2/DeployCreate2Factory.s.sol [hunk 3]
```diff
@@ -17,3 +17,17 @@ contract DeployCreate2Factory is Script {
...
   }
 }
+
+// BROADCAST
+// source .env && forge script DeployCreate2FactoryMain --skip-simulation --with-gas-price 2000000000 -vvvvv --rpc-url $ARB_MAINNET_RPC --broadcast --verify --etherscan-api-key $ARB_ETHERSCAN_API_KEY
+
+// SIMULATE
+// source .env && forge script DeployCreate2FactoryMain --with-gas-price 2000000000 -vvvvv --rpc-url $ARB_MAINNET_RPC
+
+contract DeployCreate2FactoryMain is Script {
```
Comment (pi0neerpat) [inline script/create2/DeployCreate2Factory.s.sol:27]: Does one already exist? Or its just safer to deploy it ourselves?

File: test/testlocal/nft/anvil/deployment/AnvilDeployment.t.sol [hunk 4]
```diff
@@ -58,8 +58,6 @@ abstract contract AnvilDeployment is Contracts, GoerliParams, AnvilContracts {
```
Comment (pi0neerpat) [inline test/testlocal/nft/anvil/deployment/AnvilDeployment.t.sol:58]: Do we need to update the instructions for deploying the protocol with Anvil?

**Review by pi0neerpat — APPROVED** [review]

File: script/Deploy.s.sol (hunk 1 above)
Comment (daopunk) [reply script/Deploy.s.sol:321]: Aren't we removing MAGIC?

**Pull request merged**